mysite/lti/views.py

```diff
@@ -101,21 +109,65 @@ def lti_redirect(request, course_id=None, unit_id=None):
 
     if not user.is_linked:
         user.create_links()
-
     user.login(request)
+
+    if course_ref:
+        course_id = course_ref.course.id
+    elif Role.INSTRUCTOR in roles:
+        return redirect(reverse('lti:create_courseref'))
+    else:
+        return redirect(reverse('ct:home'))
+
     user.enroll(roles, course_id)
 
     if user.is_enrolled(roles, course_id):
         # Redirect to course or unit page considering users role
         if not unit_id:
             dispatch = 'ct:course_student'
-            if roles == 'prof':
+            if roles == Role.INSTRUCTOR:
```

Comment:
> why roles and not role?

Because `roles` is a LTI param and I decide not to change the name. And in theory `roles` can be a list of roles but in practice I can't test this.